Source/WebCore/style/values/svg/StyleSVGPaint.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2025 Samuel Weinig <sam@webkit.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. ``AS IS'' AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL APPLE INC. OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+ * EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ * OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include "StyleColor.h"
+#include "StyleURL.h"
+
+namespace WebCore {
+namespace Style {
+
+enum class ForVisitedLink : bool;
+
+enum class SVGPaintType : uint8_t {
+    RGBColor,
+    None,
+    CurrentColor,
+    URINone,
+    URICurrentColor,
+    URIRGBColor,
+    URI
+};
+
+// <paint> = none | <color> | <url> [none | <color>]? | context-fill | context-stroke
+// NOTE: `context-fill` and `context-stroke` are not implemented.
+// https://svgwg.org/svg2-draft/painting.html#SpecifyingPaint
+struct SVGPaint {
+    // Models Variant<CSS::Keyword::None, Color, URL, SpaceSeparatedTuple<URL, CSS::Keyword::None>, SpaceSeparatedTuple<URL, Color>>;
```

Comment:
I guess I'm a little surprised that we don't give it an interface like this and expose the data members. With our implementation, there are invariants implied here that are not enforced.